src/main/java/com/ldtteam/structurize/placement/StructureIterators.java

```diff
@@ -0,0 +1,60 @@
+package com.ldtteam.structurize.placement;
+
+import com.ldtteam.structurize.api.util.Log;
+import com.ldtteam.structurize.placement.structure.IStructureHandler;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Class to store and handle structure iterators.
+ */
+public class StructureIterators
+{
+    /**
+     * The list of producers.
+     */
+    private static final Map<String, Function<IStructureHandler, AbstractBlueprintIterator>> iteratorProducers = new HashMap<>();
+    /*
+     * Pre-existing iterators.
+     */
+    static
+    {
+        iteratorProducers.put("default", BlueprintIteratorDefault::new);
+        iteratorProducers.put("inwardcircle", BlueprintIteratorInwardCircle::new);
+        iteratorProducers.put("inwardcircleheight1", handler -> new BlueprintIteratorInwardCircleHeight(handler, 1));
+        iteratorProducers.put("inwardcircleheight2", handler -> new BlueprintIteratorInwardCircleHeight(handler, 2));
+        iteratorProducers.put("inwardcircleheight3", handler -> new BlueprintIteratorInwardCircleHeight(handler, 3));
+        iteratorProducers.put("inwardcircleheight4", handler -> new BlueprintIteratorInwardCircleHeight(handler, 4));
+    }
+    /**
+     * Register a new producer.
+     *
+     * @param id       the id of the producer.
+     * @param producer the producer.
+     */
+    public static void registerIterator(final String id, final Function<IStructureHandler, AbstractBlueprintIterator> producer)
+    {
+        iteratorProducers.put(id, producer);
+    }
+
+    /**
+     * Get an iterator from id and with the applied structure handler.
+     *
+     * @param id      the unique id.
+     * @param handler the handler.
+     * @return the instance of the iterator.
+     */
+    public static AbstractBlueprintIterator getIterator(final String id, final IStructureHandler handler)
+    {
+        final Function<IStructureHandler, AbstractBlueprintIterator> iterator = iteratorProducers.get(id);
+        if (iterator == null)
+        {
+            Log.getLogger().warn("Could not find iterator for value:" + id + " using default instead!");
```

Comment:
yep, afaik the config value has some kind of validator in it, also I would rather crash during mod load if the value is invalid

Reply:
Crashing is always a bad idea, invalid value = default iterator + warning is good enough.

Reply:
yeah, but if you put wrong arguments to cli command then it crash, config is just neat way to have many program args
the problem is that log warning is usually not noticed by end-users, so at least send chat message on world join

Reply:
I think printing an exception here might actually be an idea. People do read and report those usually.